cranelift/codegen/src/machinst/abi_impl.rs

```diff
@@ -0,0 +1,1146 @@
+        // [MemArg::NominalSPOffset] for more details on nominal-SP tracking).
+        let stack_off = self.stackslots[slot.as_u32() as usize] as i64;
+        let sp_off: i64 = stack_off + (offset as i64);
+        M::gen_get_stack_addr(StackMemArg::NominalSPOffset(sp_off, I8), into_reg, I8)
+    }
+
+    /// Load from a spillslot.
+    fn load_spillslot(&self, slot: SpillSlot, ty: Type, into_reg: Writable<Reg>) -> Self::I {
+        // Offset from beginning of spillslot area, which is at nominal-SP + stackslots_size.
+        let islot = slot.get() as i64;
+        let spill_off = islot * 8;
+        let sp_off = self.stackslots_size as i64 + spill_off;
+        trace!("load_spillslot: slot {:?} -> sp_off {}", slot, sp_off);
+        M::gen_load_stack(StackMemArg::NominalSPOffset(sp_off, ty), into_reg, ty)
+    }
+
+    /// Store to a spillslot.
+    fn store_spillslot(&self, slot: SpillSlot, ty: Type, from_reg: Reg) -> Self::I {
+        // Offset from beginning of spillslot area, which is at nominal-SP + stackslots_size.
+        let islot = slot.get() as i64;
+        let spill_off = islot * 8;
+        let sp_off = self.stackslots_size as i64 + spill_off;
+        trace!("store_spillslot: slot {:?} -> sp_off {}", slot, sp_off);
+        M::gen_store_stack(StackMemArg::NominalSPOffset(sp_off, ty), from_reg, ty)
+    }
+
+    fn spillslots_to_stack_map(
+        &self,
+        slots: &[SpillSlot],
+        state: &<Self::I as MachInstEmit>::State,
+    ) -> StackMap {
+        let virtual_sp_offset = M::get_virtual_sp_offset_from_state(state);
+        let nominal_sp_to_fp = M::get_nominal_sp_to_fp(state);
+        assert!(virtual_sp_offset >= 0);
+        trace!(
+            "spillslots_to_stackmap: slots = {:?}, state = {:?}",
+            slots,
+            state
+        );
+        let map_size = (virtual_sp_offset + nominal_sp_to_fp) as u32;
+        let map_words = (map_size + 7) / 8;
+        let mut bits = std::iter::repeat(false)
+            .take(map_words as usize)
+            .collect::<Vec<bool>>();
+
+        let first_spillslot_word = ((self.stackslots_size + virtual_sp_offset as u32) / 8) as usize;
+        for &slot in slots {
+            let slot = slot.get() as usize;
+            bits[first_spillslot_word + slot] = true;
+        }
+
+        StackMap::from_slice(&bits[..])
+    }
+
+    fn gen_prologue(&mut self) -> Vec<Self::I> {
+        let mut insts = vec![];
+        if !self.call_conv.extends_baldrdash() {
+            // set up frame
+            insts.extend(M::gen_prologue_frame_setup().into_iter());
+        }
+
+        let mut total_stacksize = self.stackslots_size + 8 * self.spillslots.unwrap() as u32;
+        if self.call_conv.extends_baldrdash() {
+            debug_assert!(
+                !self.flags.enable_probestack(),
+                "baldrdash does not expect cranelift to emit stack probes"
+            );
+            total_stacksize += self.flags.baldrdash_prologue_words() as u32 * 8;
```

Comment:
Re the `* 8`, here and everywhere else .. this makes me nervous regarding 64/32-bit cleanness when we inevitable come to having to support a 32 bit target.  Would you consider finding all places in the diff where there is an assumption that the word size is 64-bit, and adding a warning comment line `// WARNING 32-bit unclean` or some such?  For the benefit of future porters?